utilix/batchq.py

```diff
@@ -1,265 +1,300 @@
-import subprocess
+import datetime
 import os
+import subprocess
+import re
 import tempfile
-import shlex
+from typing import Literal, Optional
+from pydantic import BaseModel, Field, validator
+from simple_slurm import Slurm
 from utilix import logger
-import getpass
-
-sbatch_template = """#!/bin/bash
-
-#SBATCH --job-name={jobname}
-#SBATCH --output={log}
-#SBATCH --error={log}
-#SBATCH --account=pi-lgrandi
-#SBATCH --qos={qos}
-#SBATCH --partition={partition}
-#SBATCH --mem-per-cpu={mem_per_cpu}
-#SBATCH --cpus-per-task={cpus_per_task}
-{node}
-{exclude_nodes}
-{hours}
-
-{job}
-"""
-
-SINGULARITY_DIR = {
-    'dali': '/dali/lgrandi/xenonnt/singularity-images',
-    'lgrandi': '/project2/lgrandi/xenonnt/singularity-images',
-    'xenon1t': '/project2/lgrandi/xenonnt/singularity-images',
-    'broadwl': '/project2/lgrandi/xenonnt/singularity-images',
-    'kicp': '/project2/lgrandi/xenonnt/singularity-images',
-    'caslake': '/project2/lgrandi/xenonnt/singularity-images',
-}
-
-SCRATCH_DIR = os.environ.get('SCRATCH')
-if len(SCRATCH_DIR.split('/'))==3 and SCRATCH_DIR.split('/')[-1]=='midway3':
-    SCRATCH_DIR += '/%s'%(os.environ['USER'])
 
+USER = os.environ.get("USER")
+SCRATCH_DIR = os.environ.get("SCRATCH", ".")
```

Comment:
For some unknown reasons, some users `SCRATCH` variable is set to be `scratch/midway3`. In order to avoid this kind of problems, can you add a sanity check if the system variables used here are in the expected form?